src/JuMP.jl

```diff
@@ -29,6 +29,12 @@ if VERSION < v"0.7-"
     const SparseArrays = Compat.SparseArrays
     const Printf = Compat.Printf
     const Sys = Compat.Sys
+    using Compat: rmul!
+    # because iteration of CartesianIndex is broken in 0.6
+    Base.Tuple(idx::CartesianIndex{1}) = (idx[1],)
+    Base.Tuple(idx::CartesianIndex{2}) = (idx[1], idx[2])
+    Base.Tuple(idx::CartesianIndex{3}) = (idx[1], idx[2], idx[3])
+    Base.Tuple(idx::CartesianIndex{4}) = (idx[1], idx[2], idx[3], idx[4])
```

Comment:
Where is this used and why do the method definitions stop at 4?

Reply:
This is because `ind2sub` is deprecated and the iteration protocol of `CartesianIndex` is broken in 0.6 because of iteration protocol change.

Sorry, I had a better plan for this but forgot to put it in.  One sec...